File: WebServer/LVSEmergency/src/main/java/it/lvsemergency/teamManagement/Team.java
package it.lvsemergency.teamManagement;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import it.lvsemergency.accountManagement.User;
import it.lvsemergency.areaInformationManagement.Area;

@Table(name = "team")
@Entity
public class Team {
	private @Id @GeneratedValue(strategy = GenerationType.IDENTITY) Integer idTeam;

	private String teamName;
	private Integer idForeman;

	@OneToMany(targetEntity = User.class, mappedBy = "idTeam", fetch = FetchType.LAZY)
	private List<User> users;
	
	@OneToOne
    @JoinColumn(name = "idArea", referencedColumnName = "idArea")
	private Area area;
	
	Team() {

	}

	public Team(Integer idTeam, String teamName, Integer idForeman, Area area) {
		super();
		this.idTeam = idTeam;
		this.teamName = teamName;
		this.idForeman = idForeman;
		this.area = area;
	}

	public Integer getIdTeam() {
		return idTeam;
	}

	public void setIdTeam(Integer idTeam) {
		this.idTeam = idTeam;
	}

	public String getTeamName() {
		return teamName;
	}

	public void setTeamName(String teamName) {
		this.teamName = teamName;
	}

	public Integer getIdForeman() {
		return idForeman;
	}

	public void setIdForeman(Integer idForeman) {
		this.idForeman = idForeman;
	}

	public List<User> getUsers() {
		return users;
	}

	public void setUsers(List<User> users) {
		this.users = users;
	}
	
	public Area getArea() {
		return area;
	}
	
	public void setArea(Area area) {
		this.area = area;
	}
}

File: Iterazioni/ClientApp/Iterazione 3/PacClientApp/src/teamManagement/team.cpp
#include "team.h"

namespace teamManagement {

Team::Team(QObject *parent)
    : helpers::EntityIF{}
{
    setParent(parent);
}

Team::Team(QJsonValue teamObject, QObject *parent)
    : helpers::EntityIF{}
{
    setParent(parent);
    fromJsonObject(teamObject.toObject());
}

QJsonDocument Team::toJsonDocument()
{
    QJsonObject teamObject;
    teamObject.insert("teamName", teamName);
    teamObject.insert("idArea", idArea);

    return QJsonDocument(teamObject);
}

void Team::fromJsonObject(QJsonObject jsonObject)
{
    idTeam = jsonObject.value("idTeam").toInt();
    teamName = jsonObject.value("teamName").toString();
    idForeman = jsonObject.value("idForeman").toInt();

    QJsonValue areaObject = jsonObject.value("area");
    area = new areaInformationManagement::Area(this);
    area->fromJsonObject(areaObject.toObject());

    QJsonValue usersObject = jsonObject.value("users");
    QJsonArray usersArray = usersObject.toArray();
    foreach (const QJsonValue & userValue, usersArray) {
        accountManagement::User *user =
                new accountManagement::User(userValue.toObject(), this);
        users.append(user);
    }
}

int Team::getIdTeam()
{
    return idTeam;
}

void Team::setIdTeam(int newIdTeam)
{
    idTeam = newIdTeam;
}

QString Team::getTeamName()
{
    return teamName;
}

void Team::setTeamName(const QString &newTeamName)
{
    teamName = newTeamName;
}

int Team::getIdForeman()
{
    return idForeman;
}

void Team::setIdForeman(int newIdForeman)
{
    idForeman = newIdForeman;
}

QList<accountManagement::User *> Team::getUsers()
{
    return users;
}

void Team::setUsers(const QList<accountManagement::User *> &newUsers)
{
    users = newUsers;
}

areaInformationManagement::Area *Team::getArea() const
{
    return area;
}

void Team::setArea(areaInformationManagement::Area *newArea)
{
    area = newArea;
}

int Team::getIdArea()
{
    return idArea;
}

void Team::setIdArea(int newIdArea)
{
    idArea = newIdArea;
}

QQmlListProperty<accountManagement::User> Team::getUsersQml()
{
    return QQmlListProperty<accountManagement::User>(this, &users);
}


}
